File: src/main/java/com/unical/travelapp/backend/catalog/dto/SingolaAttivitaDTO.java
package com.unical.travelapp.backend.catalog.dto;
import com.unical.travelapp.backend.experience.models.DTO.ImmagineResponse;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
public class SingolaAttivitaDTO {
    private Long id;
    private Long organizzatoreId;
    private String titolo;
    private String descrizione;
    private String luogo;
    private BigDecimal prezzo;
    private Integer durataMinuti;
    private Integer maxPartecipanti;
    private List<ImmagineResponse> immagini = new ArrayList<>();
    public SingolaAttivitaDTO() {
    }
    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }
    public Long getOrganizzatoreId() {
        return organizzatoreId;
    }
    public void setOrganizzatoreId(Long organizzatoreId) {
        this.organizzatoreId = organizzatoreId;
    }
    public String getTitolo() {
        return titolo;
    }
    public void setTitolo(String titolo) {
        this.titolo = titolo;
    }
    public String getDescrizione() {
        return descrizione;
    }
    public void setDescrizione(String descrizione) {
        this.descrizione = descrizione;
    }
    public String getLuogo() {
        return luogo;
    }
    public void setLuogo(String luogo) {
        this.luogo = luogo;
    }
    public BigDecimal getPrezzo() {
        return prezzo;
    }
    public void setPrezzo(BigDecimal prezzo) {
        this.prezzo = prezzo;
    }
    public Integer getDurataMinuti() {
        return durataMinuti;
    }
    public void setDurataMinuti(Integer durataMinuti) {
        this.durataMinuti = durataMinuti;
    }
    public Integer getMaxPartecipanti() {
        return maxPartecipanti;
    }
    public void setMaxPartecipanti(Integer maxPartecipanti) {
        this.maxPartecipanti = maxPartecipanti;
    }
    public List<ImmagineResponse> getImmagini() {
        return immagini;
    }
    public void setImmagini(List<ImmagineResponse> immagini) {
        this.immagini = (immagini != null) ? immagini : new ArrayList<>();
    }
}

File: travelApp_frontEnd/app/src/main/java/com/example/travelapp/data/remote/dto/SingolaAttivitaDto.kt
package com.example.travelapp.data.remote.dto

import com.example.travelapp.data.remote.ApiClient
import com.example.travelapp.domain.model.ImmagineResponse
import com.example.travelapp.domain.model.SingolaAttivita
import com.google.gson.annotations.SerializedName
import java.math.BigDecimal

data class SingolaAttivitaResponseDto(
    @SerializedName("id") val id: Long,
    @SerializedName("organizzatoreId") val organizzatoreId: Long?,
    @SerializedName("titolo") val titolo: String,
    @SerializedName("descrizione") val descrizione: String?,
    @SerializedName("luogo") val luogo: String?,
    @SerializedName("prezzo") val prezzo: BigDecimal?,
    @SerializedName("durataMinuti") val durataMinuti: Int?,
    @SerializedName("maxPartecipanti") val maxPartecipanti: Int?,
    @SerializedName("immagini") val immagini: List<ImmagineDto>? = null
) {
    fun toDomain(): SingolaAttivita = SingolaAttivita(
        id = id,
        organizzatoreId = organizzatoreId,
        titolo = titolo,
        descrizione = descrizione,
        luogo = luogo,
        prezzo = prezzo,
        durataMinuti = durataMinuti,
        maxPartecipanti = maxPartecipanti,
        immagini = immagini?.map { img ->
            ImmagineResponse(
                id = img.id,
                url = ApiClient.urlAssoluto(img.url),
                tipo = img.contentType
            )
        } ?: emptyList()
    )
}

data class SingolaAttivitaRequestDto(
    @SerializedName("titolo") val titolo: String,
    @SerializedName("descrizione") val descrizione: String?,
    @SerializedName("luogo") val luogo: String,
    @SerializedName("prezzo") val prezzo: BigDecimal,
    @SerializedName("durataMinuti") val durataMinuti: Int,
    @SerializedName("maxPartecipanti") val maxPartecipanti: Int
)